File: ios/BulletinProcessor/BulletinProcessor.m
#import <VisionCamera/FrameProcessorPlugin.h>
#import <VisionCamera/FrameProcessorPluginRegistry.h>

#if __has_include("Onnet/Onnet-Swift.h")
#import "Onnet/Onnet-Swift.h"
#else
#import "Onnet-Swift.h"
#endif

VISION_EXPORT_SWIFT_FRAME_PROCESSOR(BulletinProcessorPlugin, bulletinProcessor)

File: ios/BulletinProcessor/BulletinProcessor.swift
import VisionCamera

@objc(BulletinProcessorPlugin)
public class BulletinProcessorPlugin: FrameProcessorPlugin {
  public override init(options: [AnyHashable: Any]! = [:]) {
    super.init(options: options)
  }

  public override func callback(_ frame: Frame, withArguments arguments: [AnyHashable: Any]?) -> Any? {
    let buffer = frame.buffer
    let orientation = frame.orientation
    // code goes here
    return nil
  }
}